### PR TITLE
Added method parameter auto-filling

### PR DESCRIPTION
**Pull request by TheColorRed**
This is my first attempt to contribute to this repository, so I decided to start with something simple.

When auto completing a method, the parameters (if any) will automatically be filled in.

### CODE REVIEW

File: server/src/server.ts [hunk 1]
```diff
@@ -342,6 +342,16 @@ function addClassPropertiesMethodsParentClassesAndTraits(toReturn: CompletionIte
     }
 }
 
+function stringifyMethodParams(subNode): string
+{
+    var params = [];
+    subNode.params.forEach(param => {
+        if(param.optional){return;}
```
Comment (nevadascout) [inline server/src/server.ts]: Optional parameters should probably still be shown, but with their default value -
eg `($required, $optional = "test")`.

I _think_ this value is available under `param.value`, but you'll have to check

**Comment by nevadascout** [comment]
Cool, thanks!

I made a note about optional params (as you can see above)

**Comment by TheColorRed** [comment]
I think most editors, show multiple versions of the same function in the suggestion list.

So say you have a function with 3 parameters and the last 2 are optional, the intellisense would show three different functions like this:

```
func($one)
func($one, $two)
func($one, $two, $three)
```

Then depending on which one you choose it would pre-fill those parameters. So, if you chose the middle one it would turn this `$this->` into this `$this->func($one, $two)`

**Comment by nevadascout** [comment]
Ah okay, I misunderstood the functionality this adds

I don't think I've ever used an editor that inserts parameters as part of the method auto-completion. VS Code has is a Signature Provider (#2) which shows those three different method options as you detailed above when you're inside the brackets of a method call, which is something that I've been working on separately.

What this functionality should do is show the parameters as part of the method name, but not actually insert them. So leave `insertText` as it was before, but add the parameters to `label` including the optional ones with their values.

**Pull request merged**